Tests/xSharePoint/xSharePoint.xSPTimerJobState.Tests.ps1

```diff
@@ -0,0 +1,152 @@
+[CmdletBinding()]
+param(
+    [string] $SharePointCmdletModule = (Join-Path $PSScriptRoot "..\Stubs\SharePoint\15.0.4693.1000\Microsoft.SharePoint.PowerShell.psm1" -Resolve)
+)
+
+$ErrorActionPreference = 'stop'
+Set-StrictMode -Version latest
+
+$RepoRoot = (Resolve-Path $PSScriptRoot\..\..).Path
+$Global:CurrentSharePointStubModule = $SharePointCmdletModule 
+
+$ModuleName = "MSFT_xSPTimerJobState"
+Import-Module (Join-Path $RepoRoot "Modules\xSharePoint\DSCResources\$ModuleName\$ModuleName.psm1")
+
+Describe "xSPTimerJobState" {
+    InModuleScope $ModuleName {
+        $testParams = @{
+            Name = "job-spapp-statequery"
+            Enabled = $true
+            Schedule = "hourly between 0 and 59"
+        }
+        Import-Module (Join-Path ((Resolve-Path $PSScriptRoot\..\..).Path) "Modules\xSharePoint")
+        
+        Mock Invoke-xSharePointCommand { 
+            return Invoke-Command -ScriptBlock $ScriptBlock -ArgumentList $Arguments -NoNewScope
+        }
+                
+        Import-Module $Global:CurrentSharePointStubModule -WarningAction SilentlyContinue
+
+        Context "The server is not part of SharePoint farm" {
+            Mock Get-SPFarm { throw "Unable to detect local farm" }
+
+            It "return null from the get method" {
+                Get-TargetResource @testParams | Should Be $null
+            }
+
+            It "returns false from the test method" {
+                Test-TargetResource @testParams | Should Be $false
+            }
+
+            It "throws an exception in the set method to say there is no local farm" {
+                { Set-TargetResource @testParams } | Should throw "No local SharePoint farm was detected"
+            }
+        }
+
+        Context "The server is in a farm and the incorrect enabled settings have been applied" {
+            Mock Get-SPTimerJob {
+                $returnVal = @{
+                    Name = "job-spapp-statequery"
+                    IsDisabled = $true
+                    Schedule = "hourly between 0 and 59"
+                } 
+                return $returnVal
+            }
+            Mock Set-SPTimerJob {
+                $Global:xSharePointTimerJobUpdated = $true
+            }
+            Mock Get-SPFarm { return @{} }
+
+            It "return values from the get method" {
+                Get-TargetResource @testParams | Should Not BeNullOrEmpty
+            }
+
+            It "returns false from the test method" {
+                Test-TargetResource @testParams | Should Be $false
+            }
+
+            $Global:xSharePointTimerJobUpdated = $false
+            It "updates the timerjob settings" {
+                Set-TargetResource @testParams
+                $Global:xSharePointTimerJobUpdated | Should Be $true
```

Comment:
Here you can use Assert-MockCalled to test for the PowerShell cmdlet being called rather than the global variable approach (we only use that where we need to mock methods on objects as Pester can't mock those on its own)